pkgs/development/python-modules/mysql-connector/default.nix

```diff
@@ -1,33 +1,25 @@
-{ lib, buildPythonPackage, fetchFromGitHub
-, protobuf
+{ buildPythonPackage
+, fetchPypi
+, lib
 }:
 
 buildPythonPackage rec {
   pname = "mysql-connector";
-  version = "8.0.18";
```

Comment:
This is not OK! The latest version is replaced by a deprecated version.

Reply:
these are two different packages, the original `mysql-connector` package is really `mysql-connector-python`. https://pypi.org/search/?q=mysql-connector

Reply:
Please read the commit message, you'll get more context about the change

I'm not replacing by a deprecating version, just moving the original package to its correct location, (by inner workings of git it seems like an edit, but is a move)
```
commit 95c2389b69efed5beac9f55d2c75c7b8b418f770 (HEAD -> pythonPackages, origin/pythonPackages)
Author: Kevin Amado <kamadorueda@gmail.com>
Date:   Sat Dec 28 04:14:33 2019 -0500

    pythonPackages.mysql-connector: init at 2.2.9
    
    - There was another package called 'mysql-connector-python' that
      was using the name 'mysql-connector'
    - This is a mistake, and is fixed in this commit.
      I renamed the old 'mysql-connector' to its real name on PyPi
      'mysql-connector-python', and I created a new package
      called 'mysql-connector' (real name on PyPi)
    - References were adjusted accordingly
```

Reply:
looks like the `peewee` package needs to be updated, the only package i can find using the old package
```
[17:56:56] jon@jon-workstation ~/projects/nixpkgs (staging-next)
$ rg mysql-connector
pkgs/top-level/python-packages.nix
4177:  mysql-connector = callPackage ../development/python-modules/mysql-connector { };

pkgs/development/python-modules/peewee/default.nix
7:, withMysql ? false, mysql-connector
39:    ++ (lib.optional withMysql mysql-connector);
```

Reply:
https://pypi.org/project/mysql-connector-python/ 8.0.18
https://pypi.org/project/mysql-connector/ 2.2.9

'mysql-connector-python'@PyPi was initially named 'mysql-connector' on nixpkgs, which is not ok because there is a **real** 'mysql-connector'@PyPi that should be called 'mysql-connector'@Nixpkgs to preserve an 'intuitive naming style'

The commit fixes that

Reply:
Right, that part I did not see. 

Even so, if it is no longer maintained, I am of the opinion we should not pull it in anymore. 

Am I correct it is only an optional dependency for peewee? Python modules typically should not have optional dependencies so that could be removed.

Reply:
Fixing the attribute is OK'ish It may break some users' configuration so we typically try to avoid that, but then again we also want to avoid these kind of confusions.

Reply:
Yeah, seems like 'mysql-connector-python' is only being used as optional dependency for 'peewee', should I remove it? (in this PR, in another PR?)

Reply:
no, `mysql-connector-python` should stay. But `mysql-connector`, who would be using it?

Reply:
@FRidh, in the short-term this https://fluidattacks.com/asserts/, which I'll be adding soon

I'm just splitting the work into smaller parts for an easier peer-review, and because the community suggested it

All packages in this PR are needed by that package

(First attempt: old big PR: https://github.com/NixOS/nixpkgs/pull/76597/files)

Reply:
Yes, it's good you split it up somewhat. This PR is a good size. 

Do you know whether they're looking into using the newer connector?

Reply:
@FRidh we are already on the newer one https://gitlab.com/fluidattacks/asserts/blob/master/setup.py#L97 :)

Reply:
that's the version that is deprecated in favor of https://pypi.org/project/mysql-connector-python/

Reply:
ohhh, I see, sure, we'll move to the official one

Reply:
I understand your point, let's remove the deprecated one from this PR ?

Reply:
indeed

Reply:
done, just popped that commit and push-forced the branch